packages/keys/download-keys.ts

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env pnpm tsx
+/// <reference types="node" />
+
+import { spawn } from 'node:child_process';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { createHash } from 'node:crypto';
+import { fileURLToPath } from 'node:url';
+import { version } from './package.json';
+import { Console } from 'node:console';
+
+const stdout = new Console(process.stdout);
+const stderr = new Console(process.stderr);
+const console = stderr;
+
+const cmd = (bin: string, args: string[], opt?: { cwd?: string }) =>
+  new Promise<void>((resolve, reject) => {
+    spawn(bin, args, { stdio: 'inherit', ...opt })
+      .on('spawn', () => console.log(bin, ...args))
+      .on('error', reject)
+      .on('exit', (code, signal) => {
+        if (code) process.exit(code);
+        if (signal) process.kill(process.pid, signal);
+        resolve();
+      });
+  });
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+let workspace: string | undefined;
+workspace =
+  (await import('workspace-root').then(
+    ({ workspaceRoot }) => workspaceRoot(),
+    () => null, // package not present, or no root
+  )) || undefined;
```

Comment:
if a tool for detecting workspace-root is present, it downloads to a dir in root. this is not necessary but may be convenient for anyone else consuming this package in their own monorepo